src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosFhirDataStore.cs

```diff
@@ -352,16 +352,11 @@ public async Task<ResourceWrapper> UpdateSearchParameterIndicesAsync(ResourceWra
                 switch (exception.GetSubStatusCode())
                 {
                     case HttpStatusCode.PreconditionFailed:
+                        _logger.LogError(string.Format(Core.Resources.ResourceVersionConflict, weakETag));
                         throw new PreconditionFailedException(string.Format(Core.Resources.ResourceVersionConflict, weakETag));
 
-                    case HttpStatusCode.NotFound:
-                        throw new ResourceNotFoundException(string.Format(
-                            Core.Resources.ResourceNotFoundByIdAndVersion,
-                            resourceWrapper.ResourceTypeName,
-                            resourceWrapper.ResourceId,
-                            weakETag));
-
                     case HttpStatusCode.ServiceUnavailable:
+                        _logger.LogError(string.Format(Core.Resources.ReindexingServiceUnavailable));
```

Comment:
>string.Format(Core.Resources.ReindexingServiceUnavailable) [](http://example.com/codeflow?start=41&length=58)

string.Format(Core.Resources.ReindexingServiceUnavailable)

this looks strange. 
Why not just `Core.Resources.ReindexingServiceUnavailable`?

Reply:
Same as below! Fixed in 57a3a31.